etc/ci/upload_nightly.sh

```diff
@@ -16,25 +16,61 @@ usage() {
 
 
 upload() {
-    local nightly_filename nightly_timestamp
-    nightly_timestamp="$(date -u +"%Y-%m-%dT%H-%M-%SZ")"
-    nightly_filename="${nightly_timestamp}-$(basename "${2}")"
+    local nightly_filename
+    nightly_filename="${4}-$(basename "${2}")"
     local -r nightly_upload_dir="s3://servo-builds/nightly/${1}"
     local -r package_upload_path="${nightly_upload_dir}/${nightly_filename}"
     s3cmd --mime-type="application/octet-stream" \
           put "${2}" "${package_upload_path}"
     s3cmd cp "${package_upload_path}" "${nightly_upload_dir}/servo-latest.${3}"
 }
 
+update_brew() {
+  echo "Updating brew formula"
+
+  local package_url sha version script_dir tmp_dir nightly_filename
+
+  nightly_filename="${2}-$(basename "${1}")"
+  package_url="https://download.servo.org/nightly/macbrew/${nightly_filename}"
+  sha="$(shasum -a 256 ${1} | sed -e 's/ .*//')"
+
+  # This will transform a timestamp (2016-12-13T08-01-10Z for example)
+  # into a valid brew version number (2016.12.13).
+  version="$(echo "${2}" | \
+    sed -n 's/\([0-9]\{4\}\)-\([0-9]\{2\}\)-\([0-9]\{2\}\).*/\1.\2.\3/p')"
+
+  script_dir="${PWD}/$(dirname "${0}")"
+  tmp_dir="$(mktemp -d -t homebrew-servo)"
+  cd "${tmp_dir}"
+
+  git clone https://github.com/servo/homebrew-servo.git .
+
+  # Not using "/" as it's used in PACKAGEURL
+  sed "s|PACKAGEURL|${package_url}|g
+       s|SHA|${sha}|g
+       s|VERSION|${version}|g" \
+       < "${script_dir}/servo-binary-formula.rb.in" \
+       > Formula/servo-bin.rb
+
+  git add ./Formula/servo-bin.rb
+  git commit -m "Version bump: ${version}"
+
+  git push -qf \
+      "https://${GITHUB_HOMEBREW_TOKEN}@github.com/servo/homebrew-servo.git" \
+      master >/dev/null 2>&1
+
+  rm -rf ${tmp_dir}
```

Comment:
An alternative to using `cd` entirely is using the `-C` argument for the git invocations, e.g. `git -C "${tmp_dir}" add ...` and editing the path in the sed invocation to `"${tmp_dir}/Formula/servo-bin.rb"`. This avoids `cd` completely, which I find nicer to reason about.